x-pack/plugins/enterprise_search/public/applications/workplace_search/components/overview/overview_logic.ts

```diff
@@ -0,0 +1,204 @@
+      true,
+      {
+        setServerData: (_: IOverviewValues, { isFederatedAuth }: IOverviewValues) =>
+          isFederatedAuth,
+      },
+    ],
+    currentUser: [
+      {},
+      {
+        setServerData: (_: IOverviewValues, { currentUser }: IOverviewValues) => currentUser,
+      },
+    ],
+    fpAccount: [
+      {},
+      {
+        setServerData: (_: IOverviewValues, { fpAccount }: IOverviewValues) => fpAccount,
+      },
+    ],
+    canCreateInvitations: [
+      false,
+      {
+        setServerData: (_: IOverviewValues, { canCreateInvitations }: IOverviewValues) =>
+          canCreateInvitations,
+      },
+    ],
+    flashMessages: [
+      {},
+      {
+        setFlashMessages: (_: IOverviewValues, { flashMessages }: IOverviewValues) => flashMessages,
+      },
+    ],
+    hasUsers: [
+      false,
+      {
+        setServerData: (_: IOverviewServerData, { hasUsers }: IOverviewValues) => hasUsers,
+      },
+    ],
+    hasOrgSources: [
+      false,
+      {
+        setServerData: (_: IOverviewServerData, { hasOrgSources }: IOverviewValues) =>
+          hasOrgSources,
+      },
+    ],
+    canCreateContentSources: [
+      false,
+      {
+        setServerData: (_: IOverviewServerData, { canCreateContentSources }: IOverviewValues) =>
+          canCreateContentSources,
+      },
+    ],
+    isOldAccount: [
+      false,
+      {
+        setServerData: (_: IOverviewServerData, { isOldAccount }: IOverviewValues) => isOldAccount,
+      },
+    ],
+    sourcesCount: [
+      0,
+      {
+        setServerData: (_: IOverviewServerData, { sourcesCount }: IOverviewValues) => sourcesCount,
+      },
+    ],
+    pendingInvitationsCount: [
+      0,
+      {
+        setServerData: (_: IOverviewServerData, { pendingInvitationsCount }: IOverviewValues) =>
+          pendingInvitationsCount,
+      },
+    ],
+    accountsCount: [
+      0,
+      {
+        setServerData: (_: IOverviewServerData, { accountsCount }: IOverviewValues) =>
+          accountsCount,
+      },
+    ],
+    personalSourcesCount: [
+      0,
+      {
+        setServerData: (_: IOverviewServerData, { personalSourcesCount }: IOverviewValues) =>
+          personalSourcesCount,
+      },
+    ],
+    activityFeed: [
+      [],
+      {
+        setServerData: (_: IOverviewServerData, { activityFeed }: IOverviewValues) => activityFeed,
+      },
+    ],
+    dataLoading: [
+      true,
+      {
+        setServerData: () => false,
+        setHasErrorConnecting: () => false,
+      },
+    ],
+    hasErrorConnecting: [
+      false,
+      {
+        setHasErrorConnecting: (_: IOverviewServerData, { hasErrorConnecting }: IOverviewValues) =>
+          hasErrorConnecting,
+      },
+    ],
+  }),
+  selectors: ({ selectors }: { selectors: IOverviewValues }) => ({
+    hideOnboarding: [
+      () => [
+        selectors.hasUsers,
+        selectors.hasOrgSources,
+        selectors.isOldAccount,
+        selectors.organization,
+      ],
```

Comment:
Hm, looking at `hideOnboarding` as well - I don't think we're using this either, it's being internally within `overview.tsx`. TBH, if we're not using it in multiple places/files, I don't personally see the need to set this in Kea vs defining it within `overview.tsx`. Can we remove it or do you anticipate it being used in more than just one file?

```suggestion
```